src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -85,6 +110,9 @@ typeSearch
   -- ^ The typechecker state
   -> P.Type
   -- ^ The type we are looking for
-  -> Map (P.Qualified P.Ident) P.Type
+  -> ([(P.Qualified P.Ident, P.Type)], Maybe [(Label, Type)])
```

Comment:
That's why I chose to not include these results in the error message, but they are very useful in an editor setting:

`?x {x: {y: 1}} :: Int` would still show `_.x` which would then lead to `_.y`